plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/editor/context/project/ProjectContextProvider.kt

```diff
@@ -130,7 +129,7 @@ class ProjectContextProvider(val project: Project, private val encoderServer: En
     }
 
     private fun initEncryption(): Boolean {
-        logger.info { "project context: init key for ${project.guessProjectDir()} on port ${encoderServer.port}" }
+        logger.info { "project context: init key for ${project.basePath} on port ${encoderServer.port}" }
```

Comment:
```
  /**
   * Returns path to a directory under which project configuration files are stored ({@code .ipr} file or {@code .idea} directory). Note that it
   * is not always the direct parent of {@code .idea} directory, it may be its grand-grand parent.<br/>
   * Returns {@code null} for default project.
   * <p>It's <b>strongly recommended</b> to use other methods instead of this one (see {@link #getBaseDir()} for alternatives. Most
   * probably any use of this method in production code may lead to unexpected results for some projects (e.g. if {@code .idea} directory is
   * stored not near the project files).
   * </p>
   */
```

Reply:
True but in my sandbox, the basePath is always much superior than guessProjectDir(). When I run it in the BC projects, it can resolve to correct path.

Reply:
Fixed in latest commit.